.github/workflows/ci.yml

```diff
@@ -0,0 +1,158 @@
+      - name: install dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile --prefer-offline
+      - name: setup AWS
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: set $SHA7
+        run: echo "SHA7=$(git rev-parse --short ${{ github.event.pull_request.head.sha || github.sha }})" >> $GITHUB_ENV
+      - name: set $URL_STORYBOOK
+        run: echo "URL_STORYBOOK="https://${SHA7}.ods.dev"" >> $GITHUB_ENV
+      - name: set $COMMIT_MSG
+        run: |
+          echo "COMMIT_MSG<<EOF" >> $GITHUB_ENV
+          git log --format=%B -n 1 ${{ github.event.after }} >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+      - name: build storybook
+        run: |
+          yarn workspace @okta/odyssey-react build-storybook
+      - name: create deployment
+        uses: bobheadxi/deployments@v0.4.3
+        id: deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: preview
+      - name: deploy
+        run: |
+          aws s3 sync ./packages/odyssey-react/storybook-static/ s3://ods.dev/$SHA7 --delete
+      - name: "inform slack of deployment (#odyssey-deployments)"
+        run: bash ./scripts/notify-slack.sh
+        env:
+          INCOMING_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+          AUTHOR_NAME: ${{github.actor}}
+          BRANCH_NAME: ${{github.head_ref}}
+          PULL_REQUEST_ID: ${{github.event.pull_request.number}}
+      - name: update deployment status
+        uses: bobheadxi/deployments@v0.4.3
+        if: always()
+        with:
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+          env_url: ${{ env.URL_STORYBOOK }}
+  vrt:
+    needs: [prepare, test, deploy]
+    runs-on: ubuntu-latest
+    if: ${{ needs.prepare.outputs.vrt == 'true' }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: setup node
+        uses: actions/setup-node@v2
+        with:
+          cache: "yarn"
+          cache-dependency-path: yarn.lock
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: |
+            **/node_modules
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: lerna prepare
+        if: steps.yarn-cache.outputs.cache-hit == 'true'
+        run: lerna run prepare
+      - name: install dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile --prefer-offline
```

Comment:
why do we need to install dependencies on each of these CI steps? test/deploy/VRT. Would be good to share some knowledge around how this is setup - are we working with a docker layer cache for example and building a new container from an underlying run where dependencies are already present? Havingt to install deps 4 times on the initial build is brutal